libraries/src/AWS.Lambda.Powertools.Idempotency/Persistence/BasePersistenceStore.cs

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using AWS.Lambda.Powertools.Common;
+using AWS.Lambda.Powertools.Idempotency.Exceptions;
+using AWS.Lambda.Powertools.Idempotency.Internal;
+using AWS.Lambda.Powertools.Idempotency.Serialization;
+using DevLab.JmesPath;
+
+namespace AWS.Lambda.Powertools.Idempotency.Persistence;
+
+/// <summary>
+/// Persistence layer that will store the idempotency result.
+/// Base implementation. See <see cref="DynamoDBPersistenceStore"/> for an implementation (default one)
+/// Extend this class to use your own implementation (DocumentDB, Elasticache, ...)
+/// </summary>
+public abstract class BasePersistenceStore : IPersistenceStore
+{
+    /// <summary>
+    /// Idempotency Options
+    /// </summary>
+    private IdempotencyOptions _idempotencyOptions = null!;
+    
+    /// <summary>
+    /// Function name
+    /// </summary>
+    private string _functionName;
+    /// <summary>
+    /// Boolean to indicate whether or not payload validation is enabled
+    /// </summary>
+    protected bool PayloadValidationEnabled;
+    
+    /// <summary>
+    /// LRUCache
+    /// </summary>
+    private LRUCache<string, DataRecord> _cache = null!;
+
+    /// <summary>
+    /// Initialize the base persistence layer from the configuration settings
+    /// </summary>
+    /// <param name="idempotencyOptions">Idempotency configuration settings</param>
+    /// <param name="functionName">The name of the function being decorated</param>
+    public void Configure(IdempotencyOptions idempotencyOptions, string functionName)
+    {
+        var funcEnv = Environment.GetEnvironmentVariable(Constants.LambdaFunctionNameEnv);
+        _functionName = funcEnv ?? "testFunction";
+        if (!string.IsNullOrWhiteSpace(functionName))
+        {
+            _functionName += "." + functionName;
+        }
+        _idempotencyOptions = idempotencyOptions;
+
+        //TODO: optimize to not reconfigure
```

Comment:
What is that TODO? is that unfinished logic? if it's not the case remove it.

Reply:
@hossambarakat any idea on this one? safe to remove the TODO?

Reply:
Yup, safe to remove.